### PR TITLE
Add version bump workflow

### PR DESCRIPTION
**Pull request by yubiuser**
**PR Master branch**

**What does this PR aim to accomplish?:**

Re-do of https://github.com/pi-hole/PADD/pull/195.

Currently, we need to manually bump the version inside `padd.sh` for each release. This workflow allows to create a version bump PR automatically, if we draft a new release with a new tag. The workflow will run every hour or is triggered manually. It checks the latest tag a release has on github (includes drafts), and if the string differs from the current one set in `development` and a branch named `bump_$latest_tag` does not exist it files a PR.

See those PRs for example: https://github.com/yubiuser/PADD/pull/17, https://github.com/yubiuser/PADD/pull/18, https://github.com/yubiuser/PADD/pull/19

The idea behind this workflow is laid out here: https://github.com/pi-hole/PADD/pull/195#issuecomment-1207971448

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

### CODE REVIEW

File: .github/workflows/version_bump.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+jobs:
+  bump:
+    runs-on: ubuntu-latest
+    env:
+      REPO: ${{ github.repository }}
+      GH_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+    steps:
+      - name: Get latest release tag # can also be a draft release
+        run: |
+          LATEST_TAG=$(gh -R $REPO release list -L 1 |awk '{printf $3}')
```
Comment (PromoFaux) [inline .github/workflows/version_bump.yml:17]:
```suggestion
          LATEST_TAG=$(gh -R $REPO release list -L 1 | awk '{printf $3}')
```

Splitting hairs, I know...

Comment (yubiuser) [reply .github/workflows/version_bump.yml:17]: 😜

**Review by PromoFaux — APPROVED** [review]
Sorry, forgot to come back to this after my petty change request!

**Comment by yubiuser** [comment]
Sorry @PromoFaux 

I just noticed I had forced push without the `space` before the `awk`. Should be done now.

**Review by PromoFaux — APPROVED** [review]

**Pull request merged**